src/biz/workspace/ops.rs

```diff
@@ -131,6 +137,60 @@ pub async fn open_workspace(
   Ok(workspace)
 }
 
+pub async fn accept_workspace_invite(
+  pg_pool: &PgPool,
+  workspace_access_control: &impl WorkspaceAccessControl,
+  user_uuid: &Uuid,
+  invite_id: &Uuid,
+) -> Result<(), AppError> {
+  let mut txn = pg_pool.begin().await?;
+  update_workspace_invitation_set_invited(&mut txn, user_uuid, invite_id).await?;
+  let inv = get_invitation_by_id(&mut txn, invite_id).await?;
+  workspace_access_control
+    .insert_role(&inv.invitee_uid, &inv.workspace_id, inv.role)
+    .await?;
+  txn.commit().await?;
+  Ok(())
+}
+
+#[instrument(level = "debug", skip_all, err)]
+pub async fn invite_workspace_members(
+  pg_pool: &PgPool,
+  inviter: &Uuid,
+  workspace_id: &Uuid,
+  invitations: Vec<WorkspaceMemberInvitation>,
+) -> Result<(), AppError> {
+  let mut txn = pg_pool
+    .begin()
+    .await
+    .context("Begin transaction to invite workspace members")?;
+
+  for invitation in invitations {
+    insert_workspace_invitation(
+      &mut txn,
+      workspace_id,
+      inviter,
+      invitation.email.as_str(),
+      invitation.role,
+    )
+    .await?;
+  }
+
+  txn.commit().await?;
```

Comment:
how about providing the context("") here?

Reply:
ok, I'll add it in